src/scipp/sphinxext/__init__.py

```diff
@@ -0,0 +1,68 @@
+import re
+from typing import List, Optional
+
+PLOT_PATTERN = re.compile(r'\.plot\(')
+PLOT_DIRECTIVE_PATTERN = re.compile(r'^\s*\.\. plot::.*')
+
+
+def previous_nonempty_line(lines: List[str], index: int) -> Optional[str]:
+    index -= 1
+    while index >= 0 and not lines[index].strip():
+        index -= 1
+    return lines[index] if index >= 0 else None
+
+
+def indentation_of(s: str) -> int:
+    return len(s) - len(s.lstrip())
+
+
+def process_block(lines: List[str], begin: int, end: int) -> List[str]:
+    block = lines[begin:end]
+    if PLOT_PATTERN.search(block[-1]) is not None:
+        prev = previous_nonempty_line(lines, begin)
+        if prev and PLOT_DIRECTIVE_PATTERN.match(prev) is None:
+            # If the block is not preceded by any text or directives (e.g. comes right
+            # after the 'Examples' header), its indentation is stripped in `lines`.
+            # Adding some indentation here fixes that but also modifies indentation
+            # if the block is already indented. That should not cause problems.
+            return [
+                ' ' * indentation_of(prev) + '.. plot::', '',
+                *map(lambda l: '    ' + l, block)
+            ]
+    return block
+
+
+def is_start_of_block(line: str) -> bool:
+    return line.lstrip().startswith('>>>')
```

Comment:
I suppose this breaks (something?) when not using doctest syntax?

Reply:
It only does anything when the code uses `>>>`. I have an alternative idea using the AST directly. But it does not look like matplotlib has proper support for that.